build/simp-adapter.spec

```diff
@@ -261,23 +280,71 @@ EOM
 
 %postun
 # Post uninstall stuff
+# when $1 = 1, this is the uninstall of the previous version during an upgrade
+# when $1 = 0, this is the uninstall of the only version during an erase
+
+(
+  if [ $1 -eq 0 ]; then
+    # Previous versions of simp-adapter (0.0.6 and earlier)
+    #   1) Installed a hiera.yaml.simp file
+    #   2) Moved any initial hiera.yaml file to hiera.yaml.simpbak
+    #   3) Created a link called hiera.yaml to hiera.yaml.simp
+    #
+    # So, if this is uninstall of simp-adapter, we are going to clean up most
+    # of the residual cruft from an earlier simp-adapter, if present. However,
+    # we are intentionally going to leave 'hiera.yaml.simpbak' alone.  We
+    # could move that file back to 'hiera.yaml', if no 'hiera.yaml' exists.
+    # Alternatively, we could remove 'hiera.yaml.simpbak'.  Both of those
+    # operations seems too aggressive.  Instead, we are going to leave what
+    # may be an OBE file in place, rather than to arbitrarily change Puppet
+    # global configuration.
+    cd %{puppet_confdir}
+
+    if [ -h "hiera.yaml" ] &&  [ "$(readlink hiera.yaml)" = "hiera.yaml.simp" ]; then
+      rm hiera.yaml
+    fi
+
+    rm -f hiera.yaml.simp
+  fi
+)
 
+%posttrans
+# This strange but necessary logic is require to handle the second part of a
+# simp-adapter upgrade workaround initiated in %pre for a specific use case.
+  See %pre comments for details.
 (
   cd %{puppet_confdir}
 
-  simp_overrides='hiera.yaml'
-  for file in $simp_overrides; do
-    if [ -h $file ] && [ ! -e $file ]; then
-      rm $file
-    fi
+  if [ -f "hiera.yaml.simp.rpm_upgrade_bak" ]; then
```

Comment:
Need to only move if  hiera.yaml.simp does not exist, otherwise remove the backup.  This is necessary for simp-adapter >0.1.0 upgrades for which a puppet-server upgrade is required.  In that case the original hiera.yaml.simp is not removed, and so the move is not only unnecessary, but will fail.